### PR TITLE
Introduce the default network to EDS

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**
Fixes https://github.com/istio/istio/issues/12623.

If the sidecar does not specify its network, treat it as "local". If the endpoint doesn't have network metadata, treat it as "local". Allows defining the "local" network in the MeshNetworks with its gateways to send to remote sidecars.

### CODE REVIEW

**Comment by andraxylia** [comment]
If it is a default network, why call it local? It has a totally different meaning.

File: pilot/pkg/proxy/envoy/v2/ep_filters.go [hunk 1]
```diff
@@ -36,7 +38,7 @@ func EndpointsByNetworkFilter(endpoints []endpoint.LocalityLbEndpoints, conn *Xd
 	network, found := conn.modelNode.Metadata[model.NodeMetadataNetwork]
 	if !found {
 		// Couldn't find the sidecar network, using default/local
-		network = ""
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v2/ep_filters.go]: this is not correct. it is not consistent with CDS usage and will likely lead to issues.

**Comment by vadimeisenbergibm** [comment]
This PR is not required, https://github.com/istio/istio/issues/12623#issuecomment-487482680 described how to specify the network for the local/primary cluster.

**Pull request closed without merging**